### PR TITLE
[#3675] add cockroachdb support to python installer

### PR DESCRIPTION
**Pull request by xu-hao**

### CODE REVIEW

File: scripts/irods/database_interface.py [hunk 1]
```diff
@@ -17,7 +17,10 @@ def setup_catalog(irods_config, default_resource_directory=None):
     l = logging.getLogger(__name__)
 
     with contextlib.closing(database_connect.get_database_connection(irods_config)) as connection:
-        connection.autocommit = False
+        if irods_config.catalog_database_type == "cockroachdb":
+	    connection.autocommit = True
+	else:
```
Comment (trel) [inline scripts/irods/database_interface.py]: indentation appears off-by-one

File: scripts/irods/database_connect.py [hunk 2]
```diff
@@ -186,7 +190,10 @@ def get_database_connection(irods_config):
     os.environ['ODBCSYSINI'] = '/etc'
 
     try:
-        return pypyodbc.connect(connection_string.encode('ascii'), ansi=True)
+	if irods_config.catalog_database_type == 'cockroachdb':
```
Comment (trel) [inline scripts/irods/database_connect.py]: indentation appears off-by-one

Comment (xu-hao) [reply scripts/irods/database_connect.py]: I'll change the tabs to spaces.

**Comment by xu-hao** [comment]
found another tab, will update the commit

**Comment by xu-hao** [comment]
updated

**Comment by xu-hao** [comment]
i've updated the commit. there should be no more tabs

File: scripts/irods/database_connect.py [hunk 3]
```diff
@@ -186,7 +190,10 @@ def get_database_connection(irods_config):
     os.environ['ODBCSYSINI'] = '/etc'
 
     try:
-        return pypyodbc.connect(connection_string.encode('ascii'), ansi=True)
+        if irods_config.catalog_database_type == 'cockroachdb':
+            return pypyodbc.connect(connection_string.encode('ascii'), ansi=True, autocommit = True)
+        else:
+	    return pypyodbc.connect(connection_string.encode('ascii'), ansi=True)
```
Comment (trel) [inline scripts/irods/database_connect.py]: one last indentation tweak

Comment (xu-hao) [reply scripts/irods/database_connect.py]: thanks for catching that. will update commit.

**Pull request merged**